src/Sylius/Behat/Client/ApiPlatformClient.php

```diff
@@ -38,214 +40,100 @@ public function __construct(AbstractBrowser $client, SharedStorageInterface $sha
-    }
-
-    public function isDeletionSuccessful(): bool
-    {
-        return $this->client->getResponse()->getStatusCode() === Response::HTTP_NO_CONTENT;
-    }
-
-    /** @param string|int $value */
-    public function responseHasValue(string $key, $value): bool
+    public function buildFilter(array $filters): void
     {
-        return $this->getResponseContentValue($key) === $value;
+        $this->filters = $filters;
     }
 
     /** @param string|int $value */
-    public function relatedResourceHasValue(string $resource, string $key, $value): bool
-    {
-        $this->showRelated($resource);
-
-        return $this->getResponseContentValue($key) === $value;
-    }
-
-    /** @param string|float $value */
-    public function hasItemWithValue(string $key, $value): bool
-    {
-        foreach ($this->getCollectionItems() as $resource) {
-            if ($resource[$key] === $value) {
-                return true;
-            }
-        }
-
-        return false;
-    }
-
-    public function hasItemOnPositionWithValue(int $position, string $key, string $value): bool
+    public function addRequestData(string $key, $value): void
     {
-        return $this->getCollectionItems()[$position][$key] === $value;
+        $this->request->updateContent([$key => $value]);
     }
 
-    public function hasItemWithTranslation(string $locale, string $key, string $translation): bool
+    public function updateRequestData(array $data): void
     {
-        foreach ($this->getCollectionItems() as $resource) {
-            if (
-                isset($resource['translations']) &&
-                isset($resource['translations'][$locale]) &&
-                $resource['translations'][$locale][$key] === $translation
-            ) {
-                return true;
-            }
-        }
-
-        return false;
+        $this->request->updateContent($data);
     }
 
-    private function request(string $method, string $url, array $headers, string $content = null): void
+    public function addSubResourceData(string $key, array $data): void
     {
-        $defaultHeaders = ['HTTP_ACCEPT' => 'application/ld+json'];
-        if ($this->sharedStorage->has('token')) {
-            $defaultHeaders['HTTP_Authorization'] = 'Bearer ' . $this->sharedStorage->get('token');
-        }
-
-        $this->client->request($method, $url, [], [], array_merge($defaultHeaders, $headers), $content);
+        $this->request->addSubResource($key, $data);
     }
 
-    private function getResponseContentValue(string $key)
+    public function getResponse(): Response
     {
-        $content = json_decode($this->client->getResponse()->getContent(), true);
-
-        Assert::keyExists($content, $key);
-
-        return $content[$key];
+        return $this->client->getResponse();
     }
 
-    private function mergeArraysUniquely(array $firstArray, array $secondArray): array
+    private function request(Request $request): void
```

Comment:
I'm wondering, if it won't be useful to provide an interface for our `Request` object, in case someone would like to change something in the default behaviour.